src/pages/workspace/categories/WorkspaceCategoriesSettingsPage.tsx

```diff
@@ -94,14 +102,35 @@ function WorkspaceCategoriesSettingsPage({policy, route}: WorkspaceCategoriesSet
                     shouldPlaceSubtitleBelowSwitch
                 />
                 <View style={[styles.containerWithSpaceBetween]}>
-                    {!!currentPolicy && listItems.length > 0 && canUseWorkspaceRules && (
-                        <>
-                            <View style={[styles.mh5, styles.mt2, styles.mb1]}>
-                                <Text style={[styles.headerText]}>{translate('workspace.categories.defaultSpendCategories')}</Text>
-                                <Text style={[styles.mt1, styles.lh20]}>{translate('workspace.categories.spendCategoriesDescription')}</Text>
-                            </View>
-                            <ScrollView>{listItems}</ScrollView>
-                        </>
+                    {canUseWorkspaceRules && !!currentPolicy && sections[0].data.length > 0 && (
+                        <SelectionList
+                            headerContent={
+                                <View style={[styles.mh5, styles.mt2, styles.mb1]}>
+                                    <Text style={[styles.headerText]}>{translate('workspace.categories.defaultSpendCategories')}</Text>
+                                    <Text style={[styles.mt1, styles.lh20]}>{translate('workspace.categories.spendCategoriesDescription')}</Text>
+                                </View>
+                            }
+                            sections={sections}
+                            ListItem={SpendCategorySelectorListItem}
+                            onSelectRow={(item) => {
+                                if (!item.groupID || !item.categoryID) {
+                                    return;
+                                }
+                                setIsSelectorModalVisible(true);
+                                setCategoryID(item.categoryID);
+                                setGroupID(item.groupID);
+                            }}
+                        />
+                    )}
+                    {canUseWorkspaceRules && categoryID && groupID && (
+                        <CategorySelectorModal
```

Comment:
I moved the modal here to avoid `SectionList` nesting, which was causing a lot of problems, especially scrolling issues.